### PR TITLE
initial swing at state easing

### PR DESCRIPTION
**Pull request by scottbedard**
#62

### CODE REVIEW

**Comment by antfu** [comment]
Wow! looks very promising. Looking forward for that, thank you!

File: packages/core/useStateTransition/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          <p>Transitioned number: <b>{Math.round(this.number)}</b></p>
+        </div>
+        {Docs}
+      </div>
+    )
+  },
+})
+
+storiesOf('State', module)
+  .add('useStateTransition', () => Demo as any)
```
Comment (antfu) [inline packages/core/useStateTransition/index.stories.tsx]: Maybe simply `useTransition` under `Animation` category would be more straightforward. What do you think?

File: packages/core/useStateTransition/index.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# useStateTransition
+
+Transition state between values.
```
Comment (antfu) [inline packages/core/useStateTransition/index.md]:
```suggestion
> Transition between values
```

File: packages/core/useStateTransition/index.stories.tsx (hunk 1 above)
Comment (scottbedard) [reply packages/core/useStateTransition/index.stories.tsx]: I like that better, I'll go ahead and make the. change 👍

**Comment by antfu** [comment]
Great, thanks!

**Pull request merged**